src/core/jslib.c

```diff
@@ -28,19 +28,42 @@ JS_INIT_CONSTS();
 #define JS_CONST(name, value) HEAP32[_Jsr_##name / 4] = _hiwire_intern(value);
 
 __attribute__((import_module("sentinel"), import_name("create_sentinel"))) JsVal
-Jsv_GetNull_import(void);
+create_sentinel(int);
+
+#define ERROR_SENTINEL 1
+#define NO_VALUE_SENTINEL 2
+
+EMSCRIPTEN_KEEPALIVE JsVal
+Jsv_GetError(void)
+{
+  return create_sentinel(ERROR_SENTINEL);
+}
 
 EMSCRIPTEN_KEEPALIVE JsVal
-Jsv_GetNull(void)
+Jsv_GetNoValue(void)
 {
-  return Jsv_GetNull_import();
+  return create_sentinel(NO_VALUE_SENTINEL);
 }
 
-__attribute__((import_module("sentinel"),
-               import_name("is_sentinel"))) int JsvError_Check(JsVal);
+__attribute__((
+  import_module("sentinel"),
+  import_name("sentinel_get_value"))) int sentinel_get_value(JsVal);
+
+int
+JsvError_Check(JsVal x)
+{
+  return sentinel_get_value(x) == ERROR_SENTINEL;
+}
+
+int
+JsvNoValue_Check(JsVal x)
+{
+  return sentinel_get_value(x) == NO_VALUE_SENTINEL;
+}
 
 EM_JS_NUM(int, jslib_init_js, (void), {
   JS_INIT_CONSTS();
+  _hiwire_intern("end of constants");
```

Comment:
Can we use something like ALL CAP STRING to more clearly mark this is a magic string for snapshot?

Also adding some comment would be nice.